code/lab/manager/lab_manager.cpp

```diff
@@ -371,20 +397,175 @@ void LabManager::cleanup() {
 		// Remove all beams
 		beam_delete_all();
 
+		// Properly clean up the docker object
+		deleteDockerObject();
+
 		// Remove all objects
 		obj_delete_all();
 
 		// Clean up the particles
 		particle::kill_all();
 
+		// Clean up our path mess
+		reset_ai_path_points();
+
 		// Reset lab variables
 		CurrentMode = LabMode::None;
 		CurrentObject = -1;
 		CurrentSubtype = -1;
 		CurrentClass = -1;
+		DockerDockPoint.clear();
+		DockeeDockPoint.clear();
 		CurrentPosition = vmd_zero_vector;
 		CurrentOrientation = vmd_identity_matrix;
 		ModelFilename = "";
+		Player_ship = nullptr;
+	}
+
+	Cmdline_dis_collisions = Saved_cmdline_collisions_value;
+}
+
+void LabManager::deleteDockerObject() {
+	if (DockerObject >= 0) {
+		object* obj = &Objects[CurrentObject];
+
+		while (object_is_docked(obj)) {
+			object_jettison_cargo(obj, dock_get_first_docked_object(obj), 50, true);
+		}
+
+		obj_delete(DockerObject);
+		DockerObject = -1;
+		reset_ai_path_points();
+	}
+}
+
+void LabManager::spawnDockerObject() {
+
+	deleteDockerObject();
+
+	if (DockerDockPoint.empty() || DockeeDockPoint.empty()) {
+		return;
+	}
+
+	// Check ship class index
+	if (DockerClass < 0 || DockerClass >= static_cast<int>(Ship_info.size())) {
+		mprintf(("Invalid ship class index %d\n", DockerClass));
```

Comment:
Suggestion, return right after here and avoid having to use an else to help with readability.